src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -285,12 +288,25 @@ NoZeroEscrow::visitEntry(
 
 bool
 NoZeroEscrow::finalize(
-    STTx const&,
+    STTx const& txn,
     TER const,
     XRPAmount const,
-    ReadView const&,
+    ReadView const& rv,
     beast::Journal const& j)
 {
+    // bypass this invariant check for IOU escrows
+    if (bad_ && rv.rules().enabled(featurePaychanAndEscrowForTokens) &&
+        txn.isFieldPresent(sfTransactionType))
```

Comment:
just curious, does there exist any STTx that does not have `sfTransactionType`? is it here just for safety?

Reply:
It is probably there for safety but I can confirm.